cmd/kubeconfigwriter/main.go

```diff
@@ -23,16 +23,48 @@ import (
 	"os"
 	"strings"
 
-	"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1/cluster"
+	resource "github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
 	"github.com/tektoncd/pipeline/pkg/logging"
 	"go.uber.org/zap"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
+// Resource represents a cluster configuration (kubeconfig)
+// that can be accessed by tasks in the pipeline
+type Resource struct {
```

Comment:
shall we reuse the existing `cluster.Resource` struct
https://github.com/tektoncd/pipeline/blob/a8946d42a91808b96047a1a3090237e787d5d395/pkg/apis/resource/v1alpha1/cluster/cluster_resource.go#L34